src/eca/secrets.clj

```diff
@@ -0,0 +1,280 @@
+(ns eca.secrets
+  (:require
+   [clojure.java.io :as io]
+   [clojure.java.process :as process]
+   [clojure.string :as string]
+   [eca.logger :as logger]
+   [eca.secrets.authinfo :as secrets.authinfo]
+   [eca.secrets.netrc :as secrets.netrc])
+  (:import
+   [java.io File]
+   [java.nio.file Files LinkOption]
+   [java.nio.file.attribute PosixFilePermission]))
+
+(set! *warn-on-reflection* true)
+
+(def ^:private logger-tag "[secrets]")
+
+(defn credential-file-paths
+  "Returns ordered list of credential file paths to check."
+  []
+  (let [home (System/getProperty "user.home")
+        files [".authinfo" "_authinfo" ".netrc" "_netrc"]]
+    (mapcat (fn [filename]
+              (let [path (str (io/file home filename))]
+                [(str path ".gpg") path]))
+            files)))
+
+(defn- file-exists?
```

Comment:
Please match project codestyle (^:private instead of defn-)

Reply:
I fixed it in the files I added. There are some other places that still use `defn-` though.